src/mpid/ch4/src/mpidig_part.h

```diff
@@ -78,8 +78,21 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_mpi_pready_range(int p_low, int p_high,
         int incomplete;
         MPIR_cc_decr(&cc_part[i], &incomplete);
 
+        bool do_send = is_matched && !incomplete;
+        if (!do_send) {
+            /* if it's not matched or not complete then we miss the CTS (first one or not)
+             * so we poke progress and hopefully get it */
+            MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(0).lock);
+            mpi_errno = MPIDI_progress_test_vci(0);
+            MPIR_ERR_CHECK(mpi_errno);
```

Comment:
error checking should be outside the lock, otherwise it can jump to exit without unlocking